packages/hostScriptsServices/hostScripts/recreate_dappnode.sh

```diff
@@ -0,0 +1,70 @@
+LOG_DIR="/usr/src/dappnode/logs"
+RECREATE_DAPPNODE_LOG_FILE="$DNCORE_DIR/recreate_dappnode.log"
+
+# Name of the dappmanager Docker container
+DAPPMANAGER_CONTAINER_NAME="DAppNodeCore-dappmanager.dnp.dappnode.eth"
+
+# Create logs dir if not exist
+mkdir -p "$LOG_DIR"
+
+# dappmanager docker compose file does not exist, then install DAppNode
+if [ ! -f "$DAPPMANAGER_DNCORE_FILE" ]; then
+    echo "File $DAPPMANAGER_DNCORE_FILE does not exist. Installing DAppNode." | tee "$RECREATE_DAPPNODE_LOG_FILE"
+    wget -O - https://installer.dappnode.io | sudo UPDATE=true bash 2>&1 | tee "$RECREATE_DAPPNODE_LOG_FILE" 
+    return
+fi 
+
+# dappmanager docker container does not exist, then run docker compose up to dappmanager file
+if [ ! "$(docker ps -q -f name="$DAPPMANAGER_CONTAINER_NAME")" ]; then
+    echo "Container $DAPPMANAGER_CONTAINER_NAME does not exist. Running docker compose up to dappmanager file." | tee "$RECREATE_DAPPNODE_LOG_FILE"
+    docker-compose -f "$DAPPMANAGER_DNCORE_FILE" up -d 2>&1 | tee "$RECREATE_DAPPNODE_LOG_FILE"
+    # if the previous command fails, then recreate dappnode
+    if [ $? -ne 0 ]; then
+        echo "Error while docker compose up $DAPPMANAGER_DNCORE_FILE. Recreating dappnode to latest version." | tee "$RECREATE_DAPPNODE_LOG_FILE"
+        wget -O - https://installer.dappnode.io | sudo UPDATE=true bash 2>&1 | tee "$RECREATE_DAPPNODE_LOG_FILE"
+    fi
+    return
+fi
+
+# Check the status of the container
+# Possible docker container status: created, exited, dead, created, restarting, paused and running
+STATUS=$(docker inspect --format='{{.State.Status}}' "$DAPPMANAGER_CONTAINER_NAME" 2>/dev/null)
+
+# For safety check for docker container status not expected instead of checking for running
+if [ "$STATUS" == "restarting" ]; then
+    echo "Container $DAPPMANAGER_CONTAINER_NAME is not running. Status: $STATUS. Double checking before recreating dappnode to latest version." | tee "$RECREATE_DAPPNODE_LOG_FILE"
+    counter=0
+    # check 5 times with a 5 seconds delay the status is still the same. If 3 or more times the status is the same, then recreate dappnode
+    # This way we ensure dappmanager is not updating or in an intermedium state
+    for i in {1..5}; do
+        # check the status of the container
+        STATUS=$(docker inspect --format='{{.State.Status}}' "$DAPPMANAGER_CONTAINER_NAME" 2>/dev/null)
+        # if the status is the same as before, then increase the counter
+        if [ "$STATUS" == "restarting" ]; then
+            ((counter++))
+        fi
+        sleep 5
+    done
+    # if the counter is 3 or more, then recreate dappnode
+    if [ "$counter" -ge 3 ]; then
+        echo "Container $DAPPMANAGER_CONTAINER_NAME is restarting after 25s. Status: $STATUS. Recreating dappnode to latest version." | tee "$RECREATE_DAPPNODE_LOG_FILE"
+        wget -O - https://installer.dappnode.io | sudo UPDATE=true bash 2>&1 | tee "$RECREATE_DAPPNODE_LOG_FILE"
```

Comment:
Why don't you try here doing `compose up` before reinstalling?

Reply:
I would use reinstalling as the last option. If there is a slight chance that doing compose up works I think it is worth trying before reinstalling

Reply:
Reinstalling means that you need to rely on a third party server and you also need to have Internet connection. However, doing compose up will work anyway if the image is locally found. Then if dappmanager is still not active, we can go for reinstallation